dev-client/src/components/common/FreeformTextInput.tsx

```diff
@@ -0,0 +1,38 @@
+import {FormControl, Input} from 'native-base';
+import {useCallback, useState} from 'react';
+
+type Props = {
+  validationFunc: (input: string) => Promise<null | string>;
+  placeholder?: string;
+};
+
+/**
+ * A text input that allows the user to directly enter a text string.
+ * On submit, the result is validated. If incorrect, an error message
+ * is displayed.
+ */
+export const FreeformTextInput = ({validationFunc, placeholder}: Props) => {
+  const [hasError, setHasError] = useState<null | string>(null);
+  const [textValue, setTextValue] = useState<string>('');
+
+  const handleSubmit = useCallback(async () => {
+    const validationResults = await validationFunc(textValue);
+    if (validationResults !== null) {
+      setHasError(validationResults);
+    } else {
+      setTextValue('');
+      setHasError(null);
+    }
+  }, [validationFunc, textValue, setTextValue]);
+
+  return (
+    <FormControl isInvalid={hasError !== null}>
+      <Input
```

Comment:
nit: the linter complains about "Empty components are self-closing" here. i guess we're missing a settings somewhere to make the CI lint check always error on warnings? we should find/update that CI setting